design/httpproxy-include-route-duplicates-design.md

```diff
@@ -0,0 +1,301 @@
+      reason: DuplicateMatchConditions
+      status: "True"
+      type: IncludeError
+  currentStatus: invalid
+  description: Valid HTTPProxy
+  ...
+```
+
+Similarly, for the following root and children:
+
+```
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: example-root
+spec:
+  virtualhost:
+    fqdn: example.com
+  includes:
+  - name: example-child1
+    conditions:
+    - prefix: /foo
+  - name: example-child2
+    conditions:
+    - prefix: /
+---
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: example-child1
+spec:
+  routes:
+  - conditions:
+    - prefix: /
+    services:
+    - name: s1
+      port: 8080
+---
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: example-child2
+spec:
+  routes:
+  - conditions:
+    - prefix: /foo
+    services:
+    - name: s2
+      port: 8080
+```
+
+`example-child2` will still be marked as a duplicate and invalid since the fully qualified match `example.com/foo` is realized by the route that is configured in both children.
+
+
+**Note: The examples above all use path prefixes, but this applies to all other match condition types as well (header, query, etc.)**
+
+### Reporting Duplicates via Status
+
+## Alternatives Considered
+
+### Do not set error/warning conditions on HTTPProxies with duplicate routes
+Gateway API's HTTPRouteRule matches field defines it's semantics for matching precedence [here](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPRouteRule).
+In this API, there is no Status Condition set or error signalled to the user when there are duplicate route matches.
+The documentation lays out the expectations on conflict resolution between different route resources and within an HTTPRoute.
+We could take a similar documentation-only approach to route matching precedence, but it might leave some users with invalid configuration, confusion, etc.
+This would make it simpler to report status as writing complex header/query/path match conditions in status messages can be unwieldy.
+
+Note: this specification may change upstream in Gateway API, currently there are no conformance tests for it so as those are added we may see changes in this area.
+
+### Error condition messages without match conditions
+Instead of trying to enumerate exactly which set of match conditions caused the duplicate in the error message, we could use the json path syntax of the route to do so.
+For example we could have an error message such as `Match conditions for spec.routes[2] are a duplicate of spec.routes[0]` for errors within an HTTPProxy.
+Or: `Match conditions for spec.routes[2] are a duplicate of HTTPRoute foo/bar spec.routes[0]` for errors across HTTPProxies.
+
+This might lead to complications in accounting where routes came from which could get unwieldy to implement, but some combination of this and the currently proposed solution could be used.
+
+### Add a name field to routes
```

Comment:
this does seem appealing. can see a few usecases where having the name can be helpful.